### PR TITLE
[arcus] New port

### PR DESCRIPTION
**Pull request by stephenjust**
Arcus is used by CuraEngine as a library for writing to sockets.
Adding this port helps with CuraEngine's build as well as any other
projects that depend on Arcus.

UWP is not supported by this library.

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -33,6 +33,8 @@ apr:arm64-windows=fail
 # Requires ATL for ARM64 to be installed in CI
 azure-storage-cpp:arm64-windows=fail
 
+arcus:arm-uwp=fail
+arcus:x64-uwp=fail
```
Comment (PhoebeHui) [inline scripts/ci.baseline.txt:37]: Could you remove this in ci.baseline.txt, and add 'Supports: !uwp' in CONTROL file instead?

File: ports/arcus/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: arcus
+Version: 4.8.0
+Description: This library contains C++ code and Python3 bindings for creating a socket in a thread and using this socket to send and receive messages based on the Protocol Buffers library.
```
Comment (PhoebeHui) [inline ports/arcus/CONTROL:3]: Could you also add the Homepage?

File: ports/arcus/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    REF 4.8.0
+    SHA512 44db9b48ab6be08c30f2121d68197a7347eaf3ee255649969a773afbe45ec2433e2cc082aa72f6d40dad7ea28345da858471fff9a129365a4e848df8c8c07689
+    HEAD_REF master
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    set(ENABLE_STATIC ON)
+else()
+    set(ENABLE_STATIC OFF)
+endif()
```
Comment (PhoebeHui) [inline ports/arcus/portfile.cmake:15]: Could you use the following code instead?

string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" ENABLE_STATIC)

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@stephenjust, thanks for your contribution!

File: ports/arcus/portfile.cmake (hunk 3 above)
Comment (stephenjust) [reply ports/arcus/portfile.cmake]: Done

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

File: ports/arcus/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    HEAD_REF master
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" ENABLE_STATIC)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+        -DBUILD_PYTHON=OFF
```
Comment (BillyONeal) [inline ports/arcus/portfile.cmake:17]: Hmmm... the description says that part of the reason for this to exist is python bindings, but they're being disabled here?

Comment (stephenjust) [reply ports/arcus/portfile.cmake:17]: Good catch. For my own scenario the python bindings aren't useful so I didn't include that. Removed from the description for self-consistency.

**Pull request merged**